lib/pacemaker/pcmk_sched_probes.c

```diff
@@ -335,29 +335,38 @@ clear_actions_tracking_flag(pe_working_set_t *data_set)
     }
 }
 
+/*!
+ * \internal
+ * \brief Add restart orderings for any scheduled probes for a given resource
+ *
+ * \param[in] rsc       Resource whose probes should be ordered
+ * \param[in] data_set  Cluster working set
+ */
 static void
-order_first_rsc_probes(pe_resource_t * rsc, pe_working_set_t * data_set)
+add_restart_orderings_for_rsc(pe_resource_t *rsc, pe_working_set_t *data_set)
 {
-    GList *gIter = NULL;
     GList *probes = NULL;
 
-    g_list_foreach(rsc->children, (GFunc) order_first_rsc_probes, data_set);
-
+    // For collective resources, order each instance recursively
     if (rsc->variant != pe_native) {
+        g_list_foreach(rsc->children, (GFunc) add_restart_orderings_for_rsc,
+                       data_set);
         return;
     }
 
+    // Find all probes for given resource
     probes = pe__resource_actions(rsc, NULL, RSC_STATUS, FALSE);
 
-    for (gIter = probes; gIter != NULL; gIter= gIter->next) {
-        pe_action_t *probe = (pe_action_t *) gIter->data;
-        GList *aIter = NULL;
+    // Add probe restart orderings for each probe found
+    for (GList *iter = probes; iter != NULL; iter= iter->next) {
```

Comment:
Minor formatting nit while you're here - add a space between `iter` and `=`.

Reply:
Whoops